### PR TITLE
tests: use "keepalive" tmux session and unique session names

### PR DESCRIPTION
**Pull request by frasertweedale**
Let's hammer Travis with this commit and see if it looks like it
resolves the flaky tests.

### CODE REVIEW

File: test/TestUserAcceptance.hs [hunk 1]
```diff
@@ -282,16 +279,35 @@ assertRegex regex out = liftIO $ assertBool
...
 
-setUp :: IO (String, String, String)
-setUp = do
-  let sessionname = defaultSessionName
-  (testdir, testmdir) <- setUpTmuxSession sessionname >> setUpTempMaildir
-  pure (testdir, testmdir, sessionname)
+-- | Set up a test session.
+setUp :: Int -> IO Env
+setUp i = do
+  let sessionName = defaultSessionName <> show i
```
Comment (romanofski) [inline test/TestUserAcceptance.hs]: That would create sessions like `purebredtest1,2,3,4...` right. You do have the test case name, why not use that as an additional identifier? That'll help when debugging any failures.

File: test/TestUserAcceptance.hs [hunk 2]
```diff
@@ -37,20 +38,24 @@ data Condition
...
-        , testCanToggleHeaders
-        , testSetsMailToRead
-        , testErrorHandling
-        , testHelp
-        , testManageTagsOnMails
-        , testManageTagsOnThreads
-        ]
-
-testManageTagsOnMails :: TestTree
+  withResource pre post $ \_ ->
```
Comment (romanofski) [inline test/TestUserAcceptance.hs]: heh ... funny how "easy" that is to just add group specific setup and teardown functions ...

File: test/TestUserAcceptance.hs (hunk 1 above)
Comment (frasertweedale) [reply test/TestUserAcceptance.hs]: I'll do this in a separate commit.

Comment (romanofski) [reply test/TestUserAcceptance.hs]: Fair enough

Comment (frasertweedale) [reply test/TestUserAcceptance.hs]: PR: https://github.com/purebred-mua/purebred/pull/145

File: test/TestUserAcceptance.hs (hunk 2 above)
Comment (frasertweedale) [reply test/TestUserAcceptance.hs]: Yeah I was pleased with ease of discovery and ease of use of the interface.

**Pull request merged**